commit 6e60548adee0
Author: Rebecca N. Palmer <rebecca_palmer@zoho.com>
Date:   Sat Jul 21 20:04:00 2018 +0100

    Add LLVM 6.0 support
    
    LLVMContext::setDiagnosticHandler and LoopInfo::markAsRemoved
    have been renamed.
    
    Signed-off-by: Rebecca N. Palmer <rebecca_palmer@zoho.com>
    Reviewed-by: Yang Rong <rong.r.yang@intel.com>

--- backend/src/llvm/llvm_to_gen.cpp.orig	2017-10-24 06:04:48 UTC
+++ backend/src/llvm/llvm_to_gen.cpp
@@ -322,7 +322,11 @@ namespace gbe
     DataLayout DL(&mod);
     
     gbeDiagnosticContext dc;
+#if LLVM_VERSION_MAJOR >= 6
+    mod.getContext().setDiagnosticHandlerCallBack(&gbeDiagnosticHandler,&dc);
+#else
     mod.getContext().setDiagnosticHandler(&gbeDiagnosticHandler,&dc);
+#endif
 
 #if LLVM_VERSION_MAJOR * 10 + LLVM_VERSION_MINOR >= 37
     mod.setDataLayout(DL);
--- backend/src/llvm/llvm_unroll.cpp.orig	2017-10-24 06:04:48 UTC
+++ backend/src/llvm/llvm_unroll.cpp
@@ -205,7 +205,9 @@ namespace gbe {
           if (parentTripCount != 0 && currTripCount * parentTripCount > 32) {
             //Don't change the unrollID if doesn't force unroll.
             //setUnrollID(parentL, false);
-#if LLVM_VERSION_MAJOR * 10 + LLVM_VERSION_MINOR >= 38
+#if LLVM_VERSION_MAJOR >= 6
+            loopInfo.erase(parentL);
+#elif LLVM_VERSION_MAJOR * 10 + LLVM_VERSION_MINOR >= 38
             loopInfo.markAsRemoved(parentL);
 #else
             LPM.deleteLoopFromQueue(parentL);
